egs/babel_multilang/s5/local/nnet3/run_tdnn_multilingual.sh

```diff
@@ -0,0 +1,296 @@
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  input dim=$ivector_dim name=ivector
+  input dim=$feat_dim name=input
+  output name=output-tmp input=Append(-2,-1,0,1,2,ReplaceIndex(ivector, t, 0))
+
+  # please note that it is important to have input layer with the name=input
+  # as the layer immediately preceding the fixed-affine-layer to enable
+  # the use of short notation for the descriptor
+  # the first splicing is moved before the lda layer, so no splicing here
+  relu-renorm-layer name=tdnn1 input=Append(input@-2,input@-1,input,input@1,input@2,ReplaceIndex(ivector, t, 0)) dim=1024
+  relu-renorm-layer name=tdnn2 dim=1024
+  relu-renorm-layer name=tdnn3 input=Append(-1,2) dim=1024
+  relu-renorm-layer name=tdnn4 input=Append(-3,3) dim=1024
+  relu-renorm-layer name=tdnn5 input=Append(-3,3) dim=1024
+  relu-renorm-layer name=tdnn6 input=Append(-7,2) dim=1024
+  relu-renorm-layer name=tdnn_bn dim=$bnf_dim
+  # adding the layers for diffrent language's output
+EOF
+  # added separate outptut layer and softmax for all languages.
+  for lang_index in `seq 0 $[$num_langs-1]`;do
+    num_targets=`tree-info ${multi_ali_dirs[$lang_index]}/tree 2>/dev/null | grep num-pdfs | awk '{print $2}'` || exit 1;
+
+    echo " relu-renorm-layer name=prefinal-affine-lang-${lang_index} input=tdnn_bn dim=1024"
+    echo " output-layer name=output-${lang_index} dim=$num_targets max-change=1.5"
+  done >> $dir/configs/network.xconfig
+
+  steps/nnet3/xconfig_to_configs.py --xconfig-file $dir/configs/network.xconfig \
+    --config-dir $dir/configs/ \
+    --nnet-edits="rename-node old-name=output-0 new-name=output"
+
+  cat <<EOF >> $dir/configs/vars
+add_lda=false
+include_log_softmax=false
+EOF
+
+  # removing the extra output node "output-tmp" added for back-compatiblity with
+  # xconfig to config conversion.
+  nnet3-copy --edits="remove-output-nodes name=output-tmp" $dir/configs/ref.raw $dir/configs/ref.raw || exit 1;
+fi
+
+if [ $stage -le 9 ]; then
+  echo "$0: Generates separate egs dir per language for multilingual training."
+  # sourcing the "vars" below sets
+  #model_left_context=(something)
+  #model_right_context=(something)
+  #num_hidden_layers=(something)
+  . $dir/configs/vars || exit 1;
+  ivec="${multi_ivector_dirs[@]}"
+  if $use_ivector; then
+    ivector_opts=(--online-multi-ivector-dirs "$ivec")
+  fi
+  local/nnet3/prepare_multilingual_egs.sh --cmd "$decode_cmd" \
+    "${ivector_opts[@]}" \
+    --cmvn-opts "--norm-means=false --norm-vars=false" \
+    --left-context $model_left_context --right-context $model_right_context \
+    $num_langs ${multi_data_dirs[@]} ${multi_ali_dirs[@]} ${multi_egs_dirs[@]} || exit 1;
+
+fi
+
+if [ -z $megs_dir ];then
+  megs_dir=$dir/egs
+fi
+
+if [ $stage -le 10 ] && [ ! -z $megs_dir ]; then
+  echo "$0: Generate multilingual egs dir using "
+  echo "separate egs dirs for multilingual training."
+  if [ ! -z "$lang2weight" ]; then
+      egs_opts="--lang2weight '$lang2weight'"
```

Comment:
I think your solution would work just fine if lang2weight will not contain spaces, i.e. will be always written as "0.3,1,1" and not "0.3, 1, 1"